error/s2n_errno.h

```diff
@@ -271,7 +272,8 @@ extern __thread const char *s2n_debug_str;
 #define STRING__LINE__ STRING_(__LINE__)
 
 #define _S2N_DEBUG_LINE     "Error encountered in " __FILE__ " line " STRING__LINE__
-#define _S2N_ERROR( x )     do { s2n_debug_str = _S2N_DEBUG_LINE; s2n_errno = ( x ); s2n_calculate_stacktrace(); } while (0)
+#define _S2N_ERROR_EXPLICIT( x , debug_line )     do { s2n_debug_str = debug_line; s2n_errno = ( x ); if ( s2n_errno == S2N_ERR_LIBCRYPTO ) { s2n_save_crypto_error(); } else { s2n_libcrypto_error = 0; } s2n_calculate_stacktrace(); } while (0) 
```

Comment:
I'm not sure it's super clear what we're being explicit about here. Maybe something like:

```suggestion
#define _S2N_ERROR_WITH_LOCATION( x , debug_line )     do { s2n_debug_str = debug_line; s2n_errno = ( x ); if ( s2n_errno == S2N_ERR_LIBCRYPTO ) { s2n_save_crypto_error(); } else { s2n_libcrypto_error = 0; } s2n_calculate_stacktrace(); } while (0) 
```